### PR TITLE
REF: make Block.delete return a new Block

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

I lost track of the original use case that motivated this, but in general i think we're better off keeping block.values fixed.

### CODE REVIEW

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -1133,8 +1133,12 @@ def value_getitem(placement):
                 if len(val_locs) == len(blk.mgr_locs):
                     removed_blknos.append(blkno_l)
                 else:
-                    blk.delete(blk_locs)
-                    self._blklocs[blk.mgr_locs.indexer] = np.arange(len(blk))
+                    nb = blk.delete(blk_locs)
+                    blocks_tup = (
+                        self.blocks[:blkno_l] + (nb,) + self.blocks[blkno_l + 1 :]
```
Comment (mroeschke) [inline pandas/core/internals/managers.py:1138]: `blkno_l` is never negative here right?

Comment (jbrockmendel) [reply pandas/core/internals/managers.py:1138]: correct

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
https://github.com/pandas-dev/pandas/runs/4989389923?check_suite_focus=true

are these unrelated?

**Comment by jbrockmendel** [comment]
> are these unrelated?

Yes.  These were fixed in #45707

**Comment by jbrockmendel** [comment]
Nice.  Now #45352 is all that's left between us and block.values being immutable

**Pull request merged**